medicines/doc-index-updater/src/delete_manager/mod.rs

```diff
@@ -181,15 +191,31 @@ mod test {
     }
 
     #[test]
-    fn test_an_error_removes_delete_message() {
-        let state_manager = TestJobStatusClient {};
+    fn not_found_error_during_delete_removes_message_since_no_need_to_retry() {
+        let state_manager = given_a_state_manager();
+        let mut removeable_message = given_we_have_a_delete_message();
+        let error = given_an_error_has_occurred(anyhow!(
+            errors::DocumentNotFoundInIndex::for_content_id(String::from("any id"))
+        ));
 
-        let mut removable_message = given_we_have_a_delete_message();
-        let error = given_an_error_has_occurred();
+        let result = when_we_handle_the_error(&mut removeable_message, error, state_manager);
+
+        assert!(result.is_ok());
+        assert_eq!(removeable_message.is_removed, true, "Didn't remove message");
+    }
+
+    #[test]
+    fn recoverable_error_during_delete_does_not_remove_message_from_servicebus() {
+        let state_manager = given_a_state_manager();
+        let mut removeable_message = given_we_have_a_delete_message();
+        let error = given_an_error_has_occurred(anyhow!("Any other error"));
 
-        let result = when_we_handle_the_error(&mut removable_message, error, state_manager);
+        let result = when_we_handle_the_error(&mut removeable_message, error, state_manager);
 
         assert!(result.is_ok());
```

Comment:
I feel that wrapping these 2 lines in a then_xxxx() would be good as then the full GWT is clear - but it isn't ridicuous for the assert to be more forefront like this